adafruit_magtag/magtag.py

```diff
@@ -293,6 +300,50 @@ def set_text(self, val, index=0, auto_refresh=True):
         if auto_refresh:
             self.refresh()
 
+    def enter_deep_sleep(self, sleep_time):
+        """
+        Enter deep sleep and restart the program after a certain amount of time
+
+        :param float sleep_time: The amount of time to sleep in seconds
+
+        """
+        if self._alarm:
+            self.peripherals.neopixel_disable = True
+            self.peripherals.speaker_disable = True
+            pause = self._alarm.time.TimeAlarm(
+                monotonic_time=time.monotonic() + sleep_time
+            )
+            self._alarm.exit_and_deep_sleep_until_alarms(pause)
+        else:
+            raise NotImplementedError(
+                "Deep sleep not supported. Make sure you have the latest CircuitPython."
+            )
+
+    def enter_light_sleep(self, sleep_time):
+        """
+        Enter light sleep and resume the program after a certain amount of time
+
+        :param float sleep_time: The amount of time to sleep in seconds
+
+        """
+        if self._alarm:
+            neopixel_state = self.peripherals.neopixel_disable
+            self.peripherals.neopixel_disable = True
+            speaker_state = self.peripherals.speaker_disable
+            self.peripherals.speaker_disable = True
+            pause = self._alarm.time.TimeAlarm(
+                monotonic_time=time.monotonic() + sleep_time
+            )
+            self._alarm.light_sleep_until_alarms(pause)
+
+            self.peripherals.neopixel_disable = neopixel_state
```

Comment:
Should you re-send the neopixel pixel values?

Reply:
I'm not sure. I'll have to test if that's necessary.

Reply:
Yes, it appears that will be required.